tests/tt_eager/python_api_testing/unit_testing/misc/test_scaled_dot_product_attention.py

```diff
@@ -0,0 +1,102 @@
+    tt_K = tt_lib.tensor.Tensor(K, dtype).to(tt_lib.tensor.Layout.TILE).to(device)
+    tt_V = tt_lib.tensor.Tensor(V, dtype).to(tt_lib.tensor.Layout.TILE).to(device)
+    tt_attn_mask = tt_lib.tensor.Tensor(attn_mask, dtype).to(tt_lib.tensor.Layout.TILE).to(device)
+
+    tt_back = tt_lib.operations.primary.transformers.scaled_dot_product_attention(
+        tt_Q, tt_K, tt_V, tt_attn_mask, is_causal=True, program_config=program_config
+    )
+    tt_back = tt_back.cpu().to(tt_lib.tensor.Layout.ROW_MAJOR).to_torch()
+
+    gt = torch.nn.functional.scaled_dot_product_attention(Q, K, V, attn_mask, is_causal=False)
+
+    out_pass, out_pcc = comp_pcc(gt, tt_back, 0.994)
+    print(f"python vs pytorch: {out_pcc}")
+    assert out_pass
+
+
+@skip_for_grayskull("Unsupported in GS since L1 runs OOM with most configs")
+@pytest.mark.parametrize(
+    "dtype", [tt_lib.tensor.DataType.BFLOAT8_B, tt_lib.tensor.DataType.BFLOAT16], ids=["bfp8", "bf16"]
+)
+@pytest.mark.parametrize("q_chunk_size", [128, 256], ids=["q128", "q256"])
+@pytest.mark.parametrize("k_chunk_size", [128, 256], ids=["k128", "k256"])
+@pytest.mark.parametrize(
+    "b, nh, nkv, s, d",
+    (
+        [1, 8, 1, 2048, 128],  # Llama2-70B
+        [1, 16, 1, 2048, 64],  # Falcon-40B
+        [1, 71, 1, 2048, 64],  # Falcon-7B
+        [8, 8, 1, 2048, 128],  # Llama2-70B large batch
+        [1, 8, 1, 8192, 128],  # Llama2-70B large sequence
+    ),
+)
+def test_sdpa_tt(device, b, nh, nkv, s, d, q_chunk_size, k_chunk_size, dtype):
+    if (s % q_chunk_size != 0) or (s % k_chunk_size != 0):
+        pytest.skip("s must be divisible by q_chunk_size and k_chunk_size")
+    if nh == 8 and q_chunk_size == 128 and k_chunk_size == 128:
+        pytest.skip("Can cause OOM if profiling is enabled.")
+    tt_lib.device.DisablePersistentKernelCache()
+    run_test_sdpa_tt(device, b, nh, nkv, s, d, q_chunk_size, k_chunk_size, dtype)
+
+
+@skip_for_grayskull("Unsupported in GS since L1 runs OOM with most configs")
+@pytest.mark.parametrize(
+    "dtype", [tt_lib.tensor.DataType.BFLOAT8_B, tt_lib.tensor.DataType.BFLOAT16], ids=["bfp8", "bf16"]
+)
+@pytest.mark.parametrize("q_chunk_size", [128, 256], ids=["q128", "q256"])
+@pytest.mark.parametrize("k_chunk_size", [128, 256], ids=["k128", "k256"])
+@pytest.mark.parametrize(
+    "b, nh, nkv, s, d",
+    (
+        [1, 8, 1, 2048, 128],  # Llama2-70B
+        [1, 16, 1, 2048, 64],  # Falcon-40B
+        [1, 71, 1, 2048, 64],  # Falcon-7B
+    ),
+)
+def test_sdpa_tt_with_program_cache(device, b, nh, nkv, s, d, q_chunk_size, k_chunk_size, dtype, use_program_cache):
+    if (s % q_chunk_size != 0) or (s % k_chunk_size != 0):
+        pytest.skip("s must be divisible by q_chunk_size and k_chunk_size")
+    if nh == 8 and q_chunk_size == 128 and k_chunk_size == 128:
+        pytest.skip("Can cause OOM if profiling is enabled.")
+
+    for _ in range(2):
+        run_test_sdpa_tt(device, b, nh, nkv, s, d, q_chunk_size, k_chunk_size, dtype)
```

Comment:
Same comment about program cache applies here.

Reply:
In this test I run the test twice with program caching enabled, so the op should run on 2 different sets of inputs at different addresses. that should test it right